src/Xamarin.Android.Build.Tasks/Tasks/ResolveLibraryProjectImports.cs

```diff
@@ -135,13 +134,20 @@ static string GetTargetAssembly (ITaskItem assemblyName)
 		}
 
 		// Extracts library project contents under e.g. obj/Debug/[__library_projects__/*.jar | res/*/*]
+		// Extracts library project contents under e.g. obj/Debug/[lp/*.jar | res/*/*]
 		void Extract (
 				DirectoryAssemblyResolver res,
 				ICollection<string> jars,
 				ICollection<string> resolvedResourceDirectories,
 				ICollection<string> resolvedAssetDirectories,
 				ICollection<string> resolvedEnvironments)
 		{
+			// lets "upgrade" the old directory.
+			string oldPath = Path.GetFullPath (Path.Combine (OutputImportDirectory, "..", "__library_projects__"));
+			if (!OutputImportDirectory.Contains ("__library_projects__") && Directory.Exists (oldPath)) {
+				MonoAndroidHelper.SetDirectoryWriteable (oldPath);
```

Comment:
If we're going to delete `oldPath`, don't we need to make the *parent* directory of `oldPath` writable as well, so that we can delete `oldPath`?